File: codeiq/prime_number/prime_number.py
__author__ = 'kyota'
import fileinput

def get_prime_numbers(num):
  prime_numbers = []
  for i in range(2, num):
    count = 0
    print prime_numbers
    for prime in prime_numbers:
      if i%prime == 0: break
      if prime == prime_numbers[-1]:
        prime_numbers.append(prime)
    # for j in range(0, len(prime_numbers)):
    #   if i % prime_numbers[j] == 0 : break
    # if (j == len(prime_numbers)-1 and i%prime_numbers[-1]!=0):
    #     prime_numbers.append(i)
  return prime_numbers

def get_prime_num(num):
  prime_numbers = get_prime_numbers(num)
  return len(prime_numbers)

def main():
  numbers = [100000]
  # for line in fileinput.input():
  #   val = int(line)
  #   numbers.append(val)
  # print numbers

  for number in numbers:
    num = get_prime_num(number)
    print num


main()

File: codeiq/prime_number/prime_number.cpp
#include <iostream>
#include <string>
#include <array>
#include <vector>
#include <list>


std::vector<int> get_prime_numbers(int num) {
	std::vector<int> prime_numbers;
	for (int i=2; i < num; i++) {
		unsigned int count = 0;
		for (unsigned int j=0; j<prime_numbers.size(); j++) {
			if (i%prime_numbers[i] != 0) count++ ;
			if (count == prime_numbers.size() ) {
				prime_numbers.push_back(prime_numbers[i]);
			}

		}
	}
	return prime_numbers;
}

int main() {
	double val = 1;
	std::vector<int> nums;
	// while (std::cin >> val) {
	// 	std::cout << val;
	// 	nums.push_back(val);
	// }
	nums.push_back(10000);
	for (unsigned int i=0; i < nums.size(); i++) {
		std::vector<int> prime_numbers = get_prime_numbers(nums[i]);
		int num = prime_numbers.size();
		std::cout << num << std::endl;
	}
	// for (int =0; i<(int)nums.size(); i++) {
	// 	std::cout << nums[i] << std::endl;
	// }
    return 0;
}
